shap_adapool/pooling_strategies/k_word_pooling.py

```diff
@@ -1,6 +1,51 @@
+import shap
+import pickle
+import numpy as np
+import matplotlib
+import matplotlib.pyplot as plt
+from numpy.typing import NDArray, ArrayLike
+from typing import Callable, Iterable, Any
+from functools import reduce, partial
+from io import StringIO
+import regex_spm
+
+from ..initializer import init
+from ..pooler import shap_phrase_pooler, two_element_sum
+from ..token_concatenation import add_strings, kWord_concat
+from ..plotting import save_plot
+
+
 def main():
-    raise NotImplementedError()
+    k:int = int(input("Input the value for k: "))
```

Comment:
Would be easier to debug with `argparse` used, `input` will always hang and wait for STDIN, which is not necessarily available when running the debugger sessions. If you've got questions about how to use `argparse`, let me know, it's the golden standard in command-line option parsers in Python (and a built-in module).